### ISSUE TITLE
Import status tracking

### ISSUE THREAD

**Issue by xhero** [rism-ch/muscat#337]
I'm making this ticket to keep track of all issues during import

**Comment by xhero** [comment]
Record 806155134:
has 772 and 773 with $w = 806155134, making import fail

**Comment by xhero** [comment]
Record 450063424:
has 773 $w = 450063424, as above

**Comment by xhero** [comment]
During people import:

- Record 30098925: sigla is malformed (A-TAI; A-GAS; A-SGG; A-SCHA; A-LIsp; A-WEY) and record cannot import.
- Record 30063318: as above sigla malformed (A-LIsp; A-KRB; A-WIL; A-REI; A-LIpsm; A-TAI)

**Comment by HirschSt** [comment]
I will fix both self referentials in sources. 
With people I will try to fix the malformed call of siglas, but because they are pure strings they can nevertheless directing to nirvana.

**Comment by xhero** [comment]
There are come non-configured subfileds for Institution:

- Subfield 760 $0 missing in the marc configuration
- Subfield 785 $0 missing in the marc configuration

**Comment by xhero** [comment]
In Institution there is record 4250 with conflicting sigla "101", already assigned to number 1.

**Comment by xhero** [comment]
A missing subfield for People:

- Subfield 670 $w missing in the marc configuration

**Comment by xhero** [comment]
Institution: record 30000278 has a 710 pointing to itself ($030000278)

**Comment by xhero** [comment]
Duplicate institutions. These are actual duplicated with same siglum and different id. It is very severe as it will crash the related sources on opening them.

- 50000002: D-WÜsa
- 50000003: US-HOUr
- 50000004: D-HBa
- 50000005: D-FTZd
- 50000006: D-Au
- 50000007: I-Fb
- 50000008: LT-Vn
- 50000009: D-NATk
- 50000010: D-BAd
- 50000011: D-DGla
- 50000012: D-HVfmg
- 50000013: D-Fklein
- 50000014: D-SCHIFkp

**Comment by xhero** [comment]
Also the siglum of 50000001 seems quite malformed:

```
<record>
  <leader>00000ncm a2200000   4500</leader>
  <controlfield tag="001">50000001</controlfield>
  <datafield tag="110" ind1="1" ind2="0">
    <subfield code="a"/>
    <subfield code="g">PL- Kj.</subfield>
  </datafield>
</record>
```

**Comment by xhero** [comment]
Missing tags in Source import:

- Tag 005 missing in the marc configuration
- Tag 008 missing in the marc configuration
- Subfield 031 $2 missing in the marc configuration
- Subfield 594 $b missing in the marc configuration
- Subfield 594 $c missing in the marc configuration
- Subfield 710 $g missing in the marc configuration
- Tag 597 missing in the marc configuration
- Subfield 852 $q missing in the marc configuration
- Subfield 852 $u missing in the marc configuration
- Subfield 691 $c missing in the marc configuration
- Subfield 300 $b missing in the marc configuration
- Subfield 852 $0 missing in the marc configuration
- Subfield 772 $t missing in the marc configuration

For the tags 005 and 008 we know they are romoved, but for the others? are they ok like this? They will be dropped the first time the record is saved

**Comment by xhero** [comment]
In Sources, record **458002042** has an extremely malformed 700 $0, which gets erased during import:

```xml
 <datafield tag="700" ind1="1" ind2=" ">
    <subfield code="a">Strauss, Richard (1864<E2><80><93>1949), Komponist</subfield>
    <subfield code="d">1864-1949</subfield>
    <subfield code="0">(DE-588a)127376569</subfield>
    <subfield code="4">aut</subfield>
  </datafield>
```

**Comment by xhero** [comment]
The DE-588a problem appears in many more records. Here is a list of the $0 links, I'm preparing a list of the records that refer them. This list is ordered and duplicates removed, they are all 700$0 in Sources.

```xml
    <subfield code="0">(DE-588a)102199442</subfield>
    <subfield code="0">(DE-588a)116306300</subfield>
    <subfield code="0">(DE-588a)11640826X</subfield>
    <subfield code="0">(DE-588a)116593229</subfield>
    <subfield code="0">(DE-588a)11661868X</subfield>
    <subfield code="0">(DE-588a)116779608</subfield>
    <subfield code="0">(DE-588a)11696491X</subfield>
    <subfield code="0">(DE-588a)117106186</subfield>
    <subfield code="0">(DE-588a)117128554</subfield>
    <subfield code="0">(DE-588a)117311723</subfield>
    <subfield code="0">(DE-588a)11731191X</subfield>
    <subfield code="0">(DE-588a)117371041</subfield>
    <subfield code="0">(DE-588a)117402567</subfield>
    <subfield code="0">(DE-588a)117422959</subfield>
    <subfield code="0">(DE-588a)117493139</subfield>
    <subfield code="0">(DE-588a)118540041</subfield>
    <subfield code="0">(DE-588a)118561685</subfield>
    <subfield code="0">(DE-588a)118603108</subfield>
    <subfield code="0">(DE-588a)118697005</subfield>
    <subfield code="0">(DE-588a)11871578X</subfield>
    <subfield code="0">(DE-588a)118785109</subfield>
    <subfield code="0">(DE-588a)118792997</subfield>
    <subfield code="0">(DE-588a)118803581</subfield>
    <subfield code="0">(DE-588a)119001314</subfield>
    <subfield code="0">(DE-588a)119161850</subfield>
    <subfield code="0">(DE-588a)119207109</subfield>
    <subfield code="0">(DE-588a)12246690X</subfield>
    <subfield code="0">(DE-588a)127376569</subfield>
    <subfield code="0">(DE-588a)127969055</subfield>
    <subfield code="0">(DE-588a)130048313</subfield>
    <subfield code="0">(DE-588a)131916068</subfield>
    <subfield code="0">(DE-588a)134332709</subfield>
    <subfield code="0">(DE-588a)134339746</subfield>
    <subfield code="0">(DE-588a)134390555</subfield>
    <subfield code="0">(DE-588a)134417828</subfield>
    <subfield code="0">(DE-588a)134444094</subfield>
    <subfield code="0">(DE-588a)134479874</subfield>
    <subfield code="0">(DE-588a)134490444</subfield>
    <subfield code="0">(DE-588a)134509595</subfield>
    <subfield code="0">(DE-588a)134532775</subfield>
    <subfield code="0">(DE-588a)13453283X</subfield>
    <subfield code="0">(DE-588a)134544668</subfield>
    <subfield code="0">(DE-588a)134582810</subfield>
    <subfield code="0">(DE-588a)134620216</subfield>
    <subfield code="0">(DE-588a)137034946</subfield>
    <subfield code="0">(DE-588a)137727569</subfield>
    <subfield code="0">(DE-588a)139238611</subfield>
    <subfield code="0">(DE-588a)139238670</subfield>
    <subfield code="0">(DE-588a)174171323</subfield>
    <subfield code="0">(DE-588a)183820533</subfield>
<subfield code="0">(DE-588a)134532775</subfield>
```

**Comment by xhero** [comment]
[de_records.txt](https://github.com/rism-ch/muscat/files/279305/de_records.txt)
Attached is the list of the 297 records that have one or more of the $0 links mentioned above in their 700 tag.

**Comment by HirschSt** [comment]
Thanks, these are imported records with external bindings to the DNB not accessible by OracleDB; should I drop 700$0 if it contains such id's in the export dataset?

**Comment by xhero** [comment]
Last records for Source: there are 55 records that seem to miss the master ($0, $x, etc) in sone record and do not open. See attachment
[missing_masters.txt](https://github.com/rism-ch/muscat/files/279487/missing_masters.txt)

**Comment by xhero** [comment]
There are also 98 records that only exist as references from other records (they have no data). For example source 659002626 references 659002624:

```xml
  <datafield tag="773" ind1="0" ind2=" ">
    <subfield code="w">659002624</subfield>
  </datafield>
```

but 659002624 does not actually exist. See [unreferenced.txt](https://github.com/rism-ch/muscat/files/279672/unreferenced.txt)

**Comment by xhero** [comment]
I think the last important step is to fix the 5000000* records as they generate lots of noise during import. Apart from that it is starting to look ok.

**Comment by lpugin** [comment]
We still need to remove < and > (```&lt;```and ```&gt;```) in the fields (e.g. institution 16353)

**Comment by xhero** [comment]
I made a bush fix that added the -duplicate to the 5000000* during import. Not pretty, but works. I need this fixed as it creates a corollary of issues when importing, and created lots of noise. 320 c.a. records are affected and it takes me literally hour to go through all the errors to make sure nothing unwanted is happening.
The paramount priority is that all imports are without any errors

**Comment by xhero** [comment]
Comments specific to source are moved into #350
